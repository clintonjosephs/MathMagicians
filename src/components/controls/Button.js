import PropTypes from 'prop-types';

const Button = ({ text, className, handlerClick }) => (
  <input
    value={text}
    className={`${className || ''}`}
    key={text}
    type="button"
    onClick={handlerClick}
  />
);

Button.defaultProps = {
  className: '',
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  handlerClick: PropTypes.func.isRequired,
};

export default Button;
