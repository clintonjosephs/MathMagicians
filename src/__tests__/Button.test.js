import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '../components/controls/Button';

describe('Use Jest snapshots to test the button component.', () => {
    it('Button component renders properly', () => {
        const props = {
            text: "+",
            class: "orange",
        };

        const clickHandler = () => {}

        const component = renderer.create(
          <Button
            text={props.text}
            className={props.class}
            handlerClick={clickHandler}
          />
        );
        expect(component).toMatchSnapshot();
      });    
});

describe('Perform button click', () => {
    const props = {
        text: "+",
        class: "orange",
    };

    const clickHandler = jest.fn();

    render(<Button text={props.text}
            className={props.class}
            handlerClick={clickHandler} />);

    userEvent.click(screen.getByRole('button'));
    expect(clickHandler).toHaveBeenCalledTimes(1);
});
