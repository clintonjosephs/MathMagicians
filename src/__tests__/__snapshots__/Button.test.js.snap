// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use Jest snapshots to test the button component. Button component renders properly 1`] = `
<input
  className="orange"
  onClick={[Function]}
  type="button"
  value="+"
/>
`;
